src/Frontend/Page/User.hs

```diff
@@ -238,52 +240,44 @@ commonIdeaSpaceDelegations delegatee delegate = do
     pure $ catMaybes (schoolDelegation:classDelegations)
 
 -- | Delegation buttons works in a different way if the user opens
--- his/her own profile, or other users profile.
+-- his/her own profile, or other users profile.  First argument
+-- (`visiting`) is the (potential) delegatee, second argument (`visited`) the
+-- (potential) delegate.
 --
 -- For the own profile clicking on the delegation button, the delegate selection
 -- page is opened.
 --
 -- For other's profile, clicking on the delegation buttons mark the owner of
 -- the profile as the delegate of the current user.
-delegationButtons :: Monad m => User -> User -> [Delegation] -> HtmlT m ()
-delegationButtons delegatee delegate delegations = do
-    let ownProfile = isOwnProfile delegatee delegate
-        isActiveDelegation dscope =
-            if ownProfile
-                then any (\d -> d ^. delegationScope == dscope &&
-                                d ^. delegationFrom == delegatee ^. _Id) delegations
-                else Delegation dscope (delegatee ^. _Id) (delegate ^. _Id)
-                     `elem`
-                     delegations
+delegationButtons :: Monad m => User -> User -> [DelegationFull] -> HtmlT m ()
+delegationButtons visiting visited delegations = do
+    let ownProfile = isOwnProfile visiting visited
+        isActiveDelegation = isJust . activeDelegation
+        activeDelegation dscope = listToMaybe . (`filter` delegations) $
```

Comment:
I disagree.  the `Maybe` makes it explicit that we expect the list to be no longer than 1.